--- charts-original/templates/_helpers.tpl
+++ charts/templates/_helpers.tpl
@@ -89,8 +89,8 @@
     Resolve the tag for kubeScheduler.
 */}}
 {{- define "resolvedKubeSchedulerTag" -}}
-{{- if .Values.scheduler.kubeScheduler.imageTag }}
-{{- .Values.scheduler.kubeScheduler.imageTag | trim -}}
+{{- if .Values.scheduler.kubeScheduler.image.tag }}
+{{- .Values.scheduler.kubeScheduler.image.tag | trim -}}
 {{- else }}
 {{- include "strippedKubeVersion" . | trim -}}
 {{- end }}
@@ -106,3 +106,14 @@
 {{- $parts := split "+" .Capabilities.KubeVersion.Version -}}
 {{- print $parts._0 -}}
 {{- end -}}
+
+{{/*
+Get Rancher system-default-registry
+*/}}
+{{- define "system_default_registry" -}}
+{{- if .Values.global.systemDefaultRegistry -}}
+{{- printf "%s/" .Values.global.systemDefaultRegistry -}}
+{{- else -}}
+{{- "" -}}
+{{- end -}}
+{{- end -}}
